docs/source/common/weights_loading.rst

```diff
@@ -216,3 +179,175 @@ do the following:
+
+|
+
+    .. testcode::
+
+        from pytorch_lightning.callbacks import ModelCheckpoint
+
+
+        # saves a file like: my/path/sample-mnist-epoch=02-val_loss=0.32.ckpt
+        checkpoint_callback = ModelCheckpoint(
+            dirpath="my/path/",
+            filename="sample-mnist-{epoch:02d}-{val_loss:.2f}",
+        )
+
+- You can save top-K and last-K checkpoints by configuring the ``monitor`` and ``save_top_k`` argument.
+
+|
+
+    .. testcode::
+
+        from pytorch_lightning.callbacks import ModelCheckpoint
+
+
+        # saves top-K checkpoints based on "val_loss" metric
+        checkpoint_callback = ModelCheckpoint(
+            save_top_k=10,
+            monitor="val_loss",
+            mode="min",
+            dirpath="my/path/",
+            filename="sample-mnist-{epoch:02d}-{val_loss:.2f}",
+        )
+
+        # saves last-K checkpoints based on "global_step" metric
+        # make sure you log it inside your LightningModule
+        checkpoint_callback = ModelCheckpoint(
+            save_top_k=10,
+            monitor="global_step",
+            mode="max",
+            dirpath="my/path/",
+            filename="sample-mnist-{epoch:02d}-{global_step}",
+        )
+
+- When using iterative training which doesn't have an epoch, you can checkpoint at every ``N`` training steps by specifying ``every_n_training_steps=N``.
+- You can also control the interval of epochs between checkpoints using ``every_n_epochs`` between checkpoints, to avoid slowdowns.
+- By default, the ``ModelCheckpoint`` callback saves model weights, optimizer states, etc., but in case you have limited disk space or just need the model weights to be saved you can specify ``save_weights_only=True``.
+- You can checkpoint at a regular time interval using ``train_time_interval`` argument independent of the steps or epochs.
+
+|
+
+The :class:`~pytorch_lightning.callbacks.ModelCheckpoint` callback is very robust and should cover 99% of the use-cases. If you find a use-case that is not configured yet, feel free to open an issue with a feature request on GitHub
+and the Lightning Team will be happy to integrate/help integrate it.
+
+
+-----------
+
+
+***********************
+Customize Checkpointing
+***********************
+
+.. warning::
+
+    The Checkpoint IO API is experimental and subject to change.
+
+
+Lightning supports modifying the checkpointing save/load functionality through the ``CheckpointIO``. This encapsulates the save/load logic
+that is managed by the ``TrainingTypePlugin``. ``CheckpointIO`` is different from :meth:`~pytorch_lightning.core.hooks.CheckpointHooks.on_save_checkpoint`
+and :meth:`~pytorch_lightning.core.hooks.CheckpointHooks.on_load_checkpoint` methods as it determines how the checkpoint is saved/loaded rather than
```

Comment:
love the added distinction :)
(maybe "how the checkpoint is saved/loaded" --> "how the checkpoint is saved/loaded to storage") ?